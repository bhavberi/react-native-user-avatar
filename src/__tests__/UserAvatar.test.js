/* eslint max-len: 0 */

import "react-native";
import { Image } from "react-native";
import React from "react";
import renderer from "react-test-renderer";

import UserAvatar from "../";

it("should render <UserAvatar /> without anything, default values", () => {
  const tree = renderer.create(<UserAvatar />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render <UserAvatar /> with name, without src, default values", () => {
  const tree = renderer.create(<UserAvatar name="Bhav Beri" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render <UserAvatar /> with name, with src", () => {
  jest.useFakeTimers();
  const tree = renderer
    .create(
      <UserAvatar
        name="Bhav Beri"
        src="https://dummyimage.com/100x100/000/fff"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render <UserAvatar /> with name, if src is not an image", () => {
  jest.useFakeTimers();
  const tree = renderer
    .create(
      <UserAvatar
        name="Bhav Beri"
        src="https://github.com/bhavberi/react-native-user-avatar/"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should warn because size is not number", () => {
  const tree = renderer
    .create(<UserAvatar name="Bhav Beri" size="100" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render <UserAvatar /> with name, with custom component", () => {
  const tree = renderer
    .create(
      <UserAvatar
        name="Bhav Beri"
        component={
          <Image
            source={require("./images/rn_logo.png")}
            style={{ width: 50, height: 50 }}
          />
        }
        size={100}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
