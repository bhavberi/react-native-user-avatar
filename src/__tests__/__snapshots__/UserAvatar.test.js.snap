// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <UserAvatar /> with name, if src is not an image 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#1abc9c",
      },
      {
        "alignItems": "center",
        "borderColor": "transparent",
        "borderRadius": 16,
        "borderWidth": 0,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 32,
        "justifyContent": "center",
        "marginTop": -1.6,
        "width": 32,
      }
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        [
          {
            "color": "#fff",
            "fontSize": 12.8,
          },
          {},
        ]
      }
    >
      BB
    </Text>
  </View>
</View>
`;

exports[`should render <UserAvatar /> with name, with custom component 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#1abc9c",
      },
      {
        "alignItems": "center",
        "borderColor": "transparent",
        "borderRadius": 50,
        "borderWidth": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 100,
        "justifyContent": "center",
        "marginTop": -5,
        "width": 100,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/__tests__/images/rn_logo.png",
        }
      }
      style={
        {
          "height": 50,
          "width": 50,
        }
      }
    />
  </View>
</View>
`;

exports[`should render <UserAvatar /> with name, with src 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#1abc9c",
      },
      {
        "alignItems": "center",
        "borderColor": "transparent",
        "borderRadius": 16,
        "borderWidth": 0,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 32,
        "justifyContent": "center",
        "marginTop": -1.6,
        "width": 32,
      }
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        [
          {
            "color": "#fff",
            "fontSize": 12.8,
          },
          {},
        ]
      }
    >
      BB
    </Text>
  </View>
</View>
`;

exports[`should render <UserAvatar /> with name, without src, default values 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#1abc9c",
      },
      {
        "alignItems": "center",
        "borderColor": "transparent",
        "borderRadius": 16,
        "borderWidth": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 32,
        "justifyContent": "center",
        "marginTop": -1.6,
        "width": 32,
      }
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        [
          {
            "color": "#fff",
            "fontSize": 12.8,
          },
          {},
        ]
      }
    >
      BB
    </Text>
  </View>
</View>
`;

exports[`should render <UserAvatar /> without anything, default values 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#1abc9c",
      },
      {
        "alignItems": "center",
        "borderColor": "transparent",
        "borderRadius": 16,
        "borderWidth": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 32,
        "justifyContent": "center",
        "marginTop": -1.6,
        "width": 32,
      }
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        [
          {
            "color": "#fff",
            "fontSize": 12.8,
          },
          {},
        ]
      }
    >
      BB
    </Text>
  </View>
</View>
`;

exports[`should warn because size is not number 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#1abc9c",
      },
      {
        "alignItems": "center",
        "borderColor": "transparent",
        "borderRadius": 50,
        "borderWidth": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 100,
        "justifyContent": "center",
        "marginTop": -5,
        "width": 100,
      }
    }
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        [
          {
            "color": "#fff",
            "fontSize": 40,
          },
          {},
        ]
      }
    >
      BB
    </Text>
  </View>
</View>
`;
